app/src/main/java/com/github/multimatum_team/multimatum/model/DeadlineAdapter.kt

```diff
@@ -75,17 +92,18 @@ class DeadlineAdapter(private val context: Context) : BaseAdapter() {
                 detailTextView.setTextColor(Color.GREEN)
                 detailTextView.setTypeface(null, Typeface.BOLD)
             }
-            deadline.isDue -> {
+            clockService.now() > deadline.date -> {
                 detail = context.getString(R.string.isAlreadyDue)
             }
             else -> {
+                val timeRemaining = Period.between(clockService.now(), deadline.date)
                 detail =
-                    context.getString(R.string.DueInXDays, deadline.timeRemaining!!.days.toString())
+                    context.getString(R.string.DueInXDays, timeRemaining.days.toString())
                 // If the remaining days is too small, put them in red or orange
-                if ((deadline.timeRemaining!!.days) < URGENT) {
+                if (timeRemaining.days < URGENT) {
                     detailTextView.setTextColor(Color.RED)
                     detailTextView.setTypeface(null, Typeface.BOLD)
-                } else if ((deadline.timeRemaining!!.days) < PRESSING) {
+                } else if (timeRemaining.days < PRESSING) {
                     detailTextView.setTextColor(Color.rgb(255, 165, 0))
                     detailTextView.setTypeface(null, Typeface.BOLD)
```

Comment:
Maybe we could add PRESSING and URGENT, as deadline state in the deadline model. Then we could set a function getState to deadline class and use this instead of constant defined in the adapter (it seems bad to me to define constant in an adapter)

Reply:
The issue is that you'd have to update all deadlines from the database every day to change the states to pressing or urgent. I think pressing and urgent are not really deadline states, it just controls how they are displayed on the view.